### PR TITLE
Add a keyword to declare injective symbols

### PR DESCRIPTION
**Pull request by francoisthire**
This PR adds a keyword to declare an injective symbol. This feature is needed to type check the following example for example:

```
type : Type.

inj eta : type -> Type.

prod : A : type -> (eta A -> type) -> type.

lam : A : type -> B : (eta A -> type) -> (a:eta A -> eta (B a)) -> eta (prod A B).

def app : A : type -> B : (eta A -> type) -> eta (prod A B) -> a:eta A -> eta (B a).

def get_context : A : type -> B : type -> eta A -> eta A -> eta B -> eta A.

[A,AB,C,f,a,f',z] get_context _ C (app A AB f a) (app A AB f' a) z -->
                       (app A AB (get_context (prod A AB) C f f' z) a).
```

By the way, injective symbols already exist in lambdapi. Right now, checking that a symbol is indeed injective relies on the user.

### CODE REVIEW

**Comment by francoisthire** [comment]
According to the meeting this morning, is this PR ready to be merged @Gaspi  @GuillaumeGen ?

File: kernel/env.ml [hunk 1]
```diff
@@ -52,7 +52,7 @@ let _define (l:loc) (id:ident) (te:term) (ty_opt:typ option) : unit =
   match ty with
   | Kind -> raise (DefineExn (l,id))
   | _ ->
-    _declare l id Signature.Definable ty;
+    _declare l id Signature.Injective ty;
```
Comment (GuillaumeGen) [inline kernel/env.ml]: Why should a symbol declare by `:=` be injective?
I think we should let `Signature.Definable` here.

Comment (francoisthire) [reply kernel/env.ml]: Should be fixed now.

**Comment by GuillaumeGen** [comment]
It seems to me that now it works. However, documentation is missing, especially to mention that no checks is done about injectivity.

**Comment by francoisthire** [comment]
True, I will add that.

Le jeu. 13 sept. 2018 à 14:04, GuillaumeGen <notifications@github.com> a
écrit :

**Comment by francoisthire** [comment]
@Gaspi @GuillaumeGen Do we plan to include this in the next release?

**Pull request merged**